### PR TITLE
Remove unnecessary parentheses from method calls

### PR DESCRIPTION
**Pull request by emmabeynon**
Typically in Ruby we don't have parentheses at the end of method calls if no
arguments are being passed so this removes them.  See
https://github.com/rubocop-hq/ruby-style-guide#method-definition-parentheses.

### CODE REVIEW

**Comment by issyl0** [comment]
I've already tried this. @richardTowers has opinions about whether this course should be "programming concepts" or "idiomatic Ruby". The argument for parentheses around method calls (and even `puts`, see #7) is that knowledge gained in these Ruby lessons becomes more portable for people learning other languages in the future (eg Python).

**Comment by emmabeynon** [comment]
Purely based on my experience in a session this morning, I did find the parentheses quite jarring and found myself explaining to the student the parentheses aren't necessary because it felt wrong to pretend that they're commonly used.  I don't think it would be a massive cognitive jump if a student were to go learn another language and discover that it requires parentheses after a method/function, and certainly no different to adapting to using a different case for naming or assigning variables in a different way.  I think it would be more confusing for a student to do this course, then do some additional self-study using different material and find there's a different way of doing things and not being sure which is correct.

**Comment by richardTowers** [comment]
Yeah, it's not great confusing the coaches with unnecessary parentheses, and it's a good point about people picking up other ruby tutorials and thinking "this is weird, where's all the parens?".

As Issy says, my reasoning for putting parentheses everywhere was that it makes Ruby look a lot more like other languages (particularly python and javascript).

There are also some syntax edge cases where you do need brackets, which I think can be really confusing for learners, but it's possible these wouldn't come up in a beginners lesson.

Since it's come up a couple of times I'm inclined to agree that removing the unnecessary parens is a good idea.

I've added a 👍 👎  vote on the issue so people can add their opinions.

File: source/lesson-2-ruby-and-the-command-line.html.md.erb [hunk 1]
```diff
@@ -337,15 +337,15 @@ again. It would be nice if we could give a particular task a name, and run it
...
+irb(main):025:0> rand
 => 0.11487911496307956
 ```
 
 If we want to generate a random number less than a particular number, we can
-provide our maximum to `rand()` as an "argument":
+provide our maximum to `rand` as an "argument":
 
 ```
 irb(main):026:0> rand(100)
```
Comment (AgaDufrat) [inline source/lesson-2-ruby-and-the-command-line.html.md.erb:351]: Should the parentheses also be removed here for consistency? If we do that, we should probably suggest `(rand 12) + 1` in the answer to the challenge (lines 721 and 722).

**Comment by lfdebrux** [comment]
> Yeah, it's not great confusing the coaches with unnecessary parentheses, and it's a good point about people picking up other ruby tutorials and thinking "this is weird, where's all the parens?".

As a counterpoint on confusing the coaches, as a non-Ruby dev I've felt a bit more confident having all the parentheses around making me feel reassured!

File: source/lesson-2-ruby-and-the-command-line.html.md.erb (hunk 1 above)
Comment (AlanGabbianelli) [reply source/lesson-2-ruby-and-the-command-line.html.md.erb:351]: I think the clearer option would be to have parentheses where there are arguments passed - and no parentheses where there are no arguments passed.

Yes: `rand`, `rand(5)`
No: `rand()`, `rand 5`

**Comment by tlwr** [comment]
A couple more data points:

I had a student who kept on putting spaces between parens and the method being called: `method ()` which is technically valid if it is the right-most operand, but is a confusing error message when it is not: `method ().chained()`; `nil` does not have a method `chained`. In which case the parens are actively harmful.

However I do agree that the unnecessary parens does make the knowledge more portable.

File: source/lesson-2-ruby-and-the-command-line.html.md.erb (hunk 1 above)
Comment (richardTowers) [reply source/lesson-2-ruby-and-the-command-line.html.md.erb:351]: So:

Yes: `puts("hello world")`
No: `puts "hello world`

to be consistent? Or should we special case `puts`?

**Comment by chao-xian** [comment]
I frankly think the idea of portability is a moot point if we're picking Ruby for ease of learning. If you really wanted a portable language then why aren't we teaching Javascript? Ruby is *purposefully* a joy to learn and code with. This discussion around `()` is sucking the joy... 😞

**Comment by richardTowers** [comment]
The _real_ reason for picking Ruby was "it's already installed, and node / python3 isn't", but if this is joy sucking that's not good.

8/3 in favour on the votes, so let's 🚢 this.

**Pull request merged**